### ISSUE TITLE
Model convergence issue

### ISSUE THREAD

**Issue by wang3stones** [sp-uhh/dual-path-rnn#2]
Thanks very much for sharing your code of Dual-Path-RNN. I tried to experiment with the code and it worked smoothly. However, the training fallen into local maxima a few times and stopped at 60+ epochs, 90+ epochs, etc. when I played with different rand seeds. So far I achieved SI-SNR as 10.77 as the best on wsj0-2mix corpus. Did you face such issue before? Any tips? thanks!

**Comment by dditter** [comment]
Hi! Unfortunately, I have not experienced any issues like this. The reported results were reproducible in our setup and I am quite confused that your results are far off from our training runs with wsj0-mix2 data. Maybe you could share a plot of the training loss curve of such a run? It might help to make some sense out of what you are getting.

**Comment by wang3stones** [comment]
thanks for you reply. In your code, the number of valid data entries is 400. Did you select a portion from the original valid set? I am experimenting with the original wsj0-2mix which was generated using http://www.merl.com/demos/deep-clustering/create-speaker-mixtures.zip. The valid set got 5000 entries. Does it make difference?

**Comment by dditter** [comment]
Using only 400 out of 5000 samples during training is just for speeding up the training procedure. It will only lead to a little more uncertainty in the calculation of the estimated validation set performance if the subset is representative of the full validation set. Please notice that we shuffle the dataset before selection of the subset, so it **is** representative here.

Based on this, I do not think you will obtain substantially different results if you switch from 400 to 5000 samples in the validation set.

**Comment by wang3stones** [comment]
Eventually I found the reason at my end. It is about the data I generated. When generated wsj0-mix, I mixed up the  wv1 and wv2 files in the original WSJ0 corpus so some data is a bit noisy.
https://github.com/kaituoxu/Conv-TasNet/issues/22#issuecomment-583390461

**Comment by dditter** [comment]
Great, thanks for the feedback. I will close this issue.
